pandas/core/resample.py

```diff
@@ -1557,20 +1561,23 @@ class TimeGrouper(Grouper):
         "offset",
     )
 
+    origin: TimeGrouperOrigin
```

Comment:
So the resulting `origin` has a stricter resulting type than the `origin` accepted in `__init__`?

Reply:
Yes, the code has a try/except to convert some types in `TimestampConvertibleTypes` to a `Timestamp`.